src/MSBuild/XMake.cs

```diff
@@ -1834,10 +1834,22 @@ private static BuildResult ExecuteRestore(string projectFile, string toolsVersio
                 hostServices: null,
                 flags: flags);
 
-            return ExecuteBuild(buildManager, restoreRequest);
+            bool? wasBuildCheckEnabled = null;
+            try
+            {
+                wasBuildCheckEnabled = buildManager.ConfigureBuildCheck(false);
+                BuildResult restoreResult = ExecuteBuild(buildManager, restoreRequest);
+                return restoreResult;
+            }
+            finally
+            {
+                if (wasBuildCheckEnabled == true)
+                {
+                    buildManager.ConfigureBuildCheck(true);
+                }
```

Comment:
this feels that it might not need to be conditional - if the buildcheck was not enabled, then the `_buildCheckManagerForEnabledState` is anyways the null pattern manager - and hence setting doesn't hurt

Reply:
Correct. If we go this way, I plan to clean it a bit, so far I was mostly focusing on describing the approach